### ISSUE TITLE
ppliteseg修改crop_size参数模型无法训练

### ISSUE THREAD

**Issue by gezhaoDL** [PaddlePaddle/PaddleSeg#2965]
### 问题确认 Search before asking

- [X] 我已经搜索过问题，但是没有找到解答。I have searched the question and found no related answer.

### 请提出你的问题 Please ask your question

训练自己的数据，crop_size设置为[1024, 512]训练正常，设置为[960, 720]训练出现下图报错
![image](https://user-images.githubusercontent.com/19384950/217776728-09e9471f-40fe-46e5-8fb5-6b0d0cc088ad.png)

配置对比：
1024*512：
`batch_size: 16
iters: 160000
loss:
  coef:
  - 1
  - 1
  - 1
  types:
  - ignore_index: 255
    min_kept: 520000
    type: OhemCrossEntropyLoss
  - ignore_index: 255
    min_kept: 520000
    type: OhemCrossEntropyLoss
  - ignore_index: 255
    min_kept: 520000
    type: OhemCrossEntropyLoss
lr_scheduler:
  end_lr: 0
  learning_rate: 0.00125
  power: 0.9
  type: PolynomialDecay
  warmup_iters: 1000
  warmup_start_lr: 1.0e-05
model:
  arm_out_chs:
  - 32
  - 64
  - 128
  backbone:
    in_channels: 3
    pretrained: https://bj.bcebos.com/paddleseg/dygraph/PP_STDCNet2.tar.gz
    type: STDC2
  num_classes: 25
  seg_head_inter_chs:
  - 32
  - 64
  - 64
  type: PPLiteSeg
optimizer:
  momentum: 0.9
  type: sgd
  weight_decay: 0.0005
test_config:
  aug_eval: true
  scales: 0.75
train_dataset:
  dataset_root: data/selfdata0107
  img_channels: 3
  mode: train
  num_classes: 25
  train_path: data/selfdata0107/train.txt
  transforms:
  - max_scale_factor: 1.5
    min_scale_factor: 0.125
    scale_step_size: 0.125
    type: ResizeStepScaling
  - crop_size:
    - 1024
    - 512
    type: RandomPaddingCrop
  - type: RandomHorizontalFlip
  - brightness_range: 0.5
    contrast_range: 0.5
    saturation_range: 0.5
    type: RandomDistort
  - type: Normalize
  type: Dataset
val_dataset:
  dataset_root: data/selfdata0107
  img_channels: 3
  mode: val
  num_classes: 25
  transforms:
  - type: Normalize
  type: Dataset
  val_path: data/selfdata0107/val.txt`

960*720：
`batch_size: 16
iters: 160000
loss:
  coef:
  - 1
  - 1
  - 1
  types:
  - ignore_index: 255
    min_kept: 730000
    type: OhemCrossEntropyLoss
  - ignore_index: 255
    min_kept: 730000
    type: OhemCrossEntropyLoss
  - ignore_index: 255
    min_kept: 730000
    type: OhemCrossEntropyLoss
lr_scheduler:
  end_lr: 0
  learning_rate: 0.00125
  power: 0.9
  type: PolynomialDecay
  warmup_iters: 1000
  warmup_start_lr: 1.0e-05
model:
  arm_out_chs:
  - 32
  - 64
  - 128
  backbone:
    in_channels: 3
    pretrained: https://bj.bcebos.com/paddleseg/dygraph/PP_STDCNet2.tar.gz
    type: STDC2
  num_classes: 25
  seg_head_inter_chs:
  - 32
  - 64
  - 64
  type: PPLiteSeg
optimizer:
  momentum: 0.9
  type: sgd
  weight_decay: 0.0005
test_config:
  aug_eval: true
  scales: 0.75
train_dataset:
  dataset_root: data/selfdata0107
  img_channels: 3
  mode: train
  num_classes: 25
  train_path: data/selfdata0107/train.txt
  transforms:
  - max_scale_factor: 1.5
    min_scale_factor: 0.125
    scale_step_size: 0.125
    type: ResizeStepScaling
  - crop_size:
    - 960
    - 720
    type: RandomPaddingCrop
  - type: RandomHorizontalFlip
  - brightness_range: 0.5
    contrast_range: 0.5
    saturation_range: 0.5
    type: RandomDistort
  - type: Normalize
  type: Dataset
val_dataset:
  dataset_root: data/selfdata0107
  img_channels: 3
  mode: val
  num_classes: 25
  transforms:
  - type: Normalize
  type: Dataset
  val_path: data/selfdata0107/val.txt`

**Comment by Bobholamovic** [comment]
请问这个问题可以稳定复现嘛？因为从报错信息看起来像是显存的 OOM，也有可能是前一次运行之后显存资源没有得到释放，导致第二次运行的时候显存不够用。请先排查一下是否是这个问题~

**Comment by gezhaoDL** [comment]
> 请问这个问题可以稳定复现嘛？因为从报错信息看起来像是显存的 OOM，也有可能是前一次运行之后显存资源没有得到释放，导致第二次运行的时候显存不够用。请先排查一下是否是这个问题~

可以稳定复现，显存是没有占用的，我第一次训练960x720的报错，再训练1025x512的正常，另外这是机器的配置：
`------------Environment Information-------------

platform: Linux-3.10.0-1160.71.1.el7.x86_64-x86_64-with-glibc2.17
Python: 3.9.15 (main, Nov 24 2022, 14:31:59) [GCC 11.2.0]
Paddle compiled with cuda: True
NVCC: Build cuda_11.2.r11.2/compiler.29618528_0
cudnn: 8.2
GPUs used: 1
CUDA_VISIBLE_DEVICES: 1
GPU: ['GPU 0: Tesla T4', 'GPU 1: Tesla T4']
GCC: gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44)
PaddleSeg: 2.7.0
PaddlePaddle: 2.4.1
OpenCV: 4.6.0
`

**Comment by Bobholamovic** [comment]
Tesla T4的显存是16G，相对 `batch_size=16`的设置来说不是很充足。建议检查一下输入图像尺寸为`1024*512`时显存占用是否已经接近临界值。由于`960*720=691200 > 1024*512=524288`，尺寸为`960*720`的输入图像具有更多的像素，这也会导致更大的显存占用。如果在`1024*512`的情况下，显存占用已经接近显卡能承载的极限值，那么当分辨率调整为`960*720`时很有可能出现 OOM 问题。

**Comment by gezhaoDL** [comment]
> Tesla T4的显存是16G，相对 `batch_size=16`的设置来说不是很充足。建议检查一下输入图像尺寸为`1024*512`时显存占用是否已经接近临界值。由于`960*720=691200 > 1024*512=524288`，尺寸为`960*720`的输入图像具有更多的像素，这也会导致更大的显存占用。如果在`1024*512`的情况下，显存占用已经接近显卡能承载的极限值，那么当分辨率调整为`960*720`时很有可能出现 OOM 问题。

确实是这样，减小batchsize可以训练了
